utility/src/main/java/org/oppia/android/util/logging/FeatureFlagNameToIntegerNameConverter.kt

```diff
@@ -0,0 +1,51 @@
+package org.oppia.android.util.logging
+
+import org.oppia.android.util.platformparameter.APP_AND_OS_DEPRECATION
+import org.oppia.android.util.platformparameter.DOWNLOADS_SUPPORT
+import org.oppia.android.util.platformparameter.EDIT_ACCOUNTS_OPTIONS_UI
+import org.oppia.android.util.platformparameter.ENABLE_MULTIPLE_CLASSROOMS
+import org.oppia.android.util.platformparameter.ENABLE_NPS_SURVEY
+import org.oppia.android.util.platformparameter.ENABLE_ONBOARDING_FLOW_V2
+import org.oppia.android.util.platformparameter.ENABLE_PERFORMANCE_METRICS_COLLECTION
+import org.oppia.android.util.platformparameter.EXTRA_TOPIC_TABS_UI
+import org.oppia.android.util.platformparameter.FAST_LANGUAGE_SWITCHING_IN_LESSON
+import org.oppia.android.util.platformparameter.INTERACTION_CONFIG_CHANGE_STATE_RETENTION
+import org.oppia.android.util.platformparameter.LEARNER_STUDY_ANALYTICS
+import org.oppia.android.util.platformparameter.LOGGING_LEARNER_STUDY_IDS
+import org.oppia.android.util.platformparameter.SPOTLIGHT_UI
+import javax.inject.Inject
+
+/**
+ * Converter for feature flag names to integers for more compact logging to address Google
+ * Analytics character limits. GA4 limits the characters permitted in a log event parameter value to
+ * a maximum of 100 characters as of March 2025.
+ */
+class FeatureFlagNameToIntegerNameConverter @Inject constructor() {
+  /**
+   * Converts a feature flag name to an integer.
+   *
+   * These numbers are supposed to be incremented indefinitely and the integer representation of
+   * each feature flag name should not be reused even after the feature flag is no longer in use.
+   *
+   * @param flagName the string constant flag name to convert.
+   * @return an integer representation of the flag name.
```

Comment:
```suggestion
   * @param flagName the string constant flag name to convert
   * @return an integer representation of the flag name
```

Nit: we don't end at-clauses with punctuation per style guide.

Reply:
Done